packages/slice-machine/lib/builders/SliceBuilder/index.tsx

```diff
@@ -146,4 +183,28 @@ const SliceBuilder: React.FunctionComponent = () => {
   );
 };
 
-export default SliceBuilder;
+const SliceBuilderWithRouter = () => {
+  const router = useRouter();
+
+  const { Model } = useSelector((store: SliceMachineStoreType) => ({
+    Model: selectCurrentSlice(
+      store,
+      router.query.lib as string,
+      router.query.sliceName as string
+    ),
+  }));
+
+  if (!Model) {
+    void router.replace("/");
+    return null;
+  }
+
+  return (
+    <SliceBuilder
+      initModel={Model}
```

Comment:
I wonder when do we split slice data and variation data. (slice data for like displaying the title and variation data for displaying fields for instance.

If it's even doable.

Reply:
❓ don't understand what you mean by this

Reply:
I went a little bit overboard haha, we can keep that for another time.